src/main/webapp/app/service/firebase/firebase-gene-review-service.spec.ts

```diff
@@ -0,0 +1,340 @@
+import { ReviewAction } from 'app/config/constants/firebase';
+import { DiffMethod } from 'react-diff-viewer-continued';
+import _ from 'lodash';
+
+describe('Firebase Gene Review Service', () => {
+  const DEFAULT_USERNAME = 'Test User';
+  const DEFAULT_DATE = new Date('2023-01-01');
+  const mockFirebaseRepository = mock<FirebaseRepository>();
+  const mockAuthStore = mock<AuthStore>();
+  const mockMetaService = mock<FirebaseMetaService>();
+  const mockHistoryService = mock<FirebaseHistoryService>();
+  const mockVusService = mock<FirebaseVusService>();
+  let firebaseGeneReviewService: FirebaseGeneReviewService;
+
+  const _beforeEach = () => {
+    mockReset(mockFirebaseRepository);
+    mockReset(mockAuthStore);
+    mockReset(mockMetaService);
+    mockReset(mockHistoryService);
+    mockReset(mockVusService);
+    firebaseGeneReviewService = new FirebaseGeneReviewService(
+      mockFirebaseRepository,
+      mockAuthStore,
+      mockMetaService,
+      mockHistoryService,
+      mockVusService,
+    );
+    jest.useFakeTimers().setSystemTime(DEFAULT_DATE);
+  };
+
+  describe('updateReviewableContent', () => {
+    beforeEach(() => _beforeEach());
+
+    it('should update to firebase path with correct object', async () => {
+      await firebaseGeneReviewService.updateReviewableContent(
+        'Genes/BRAF/mutations/0/description',
+        'old',
+        'new',
+        new Review(DEFAULT_USERNAME),
+        'uuid',
+      );
+
+      expect(mockFirebaseRepository.update).toHaveBeenCalledWith('Genes/BRAF', {
+        'mutations/0/description': 'new',
+        'mutations/0/description_review': {
+          lastReviewed: 'old',
+          updateTime: DEFAULT_DATE.getTime(),
+          updatedBy: mockAuthStore.fullName,
+        },
+      });
+    });
+
+    it('should update germline content', async () => {
+      await firebaseGeneReviewService.updateReviewableContent(
+        'Germline_Genes/BRAF/summary',
+        'old',
+        'new',
+        new Review(DEFAULT_USERNAME),
+        'uuid',
+      );
+
+      expect(mockFirebaseRepository.update).toHaveBeenCalledWith('Germline_Genes/BRAF', {
+        summary: 'new',
+        summary_review: {
+          lastReviewed: 'old',
+          updateTime: DEFAULT_DATE.getTime(),
+          updatedBy: mockAuthStore.fullName,
+        },
+      });
+    });
+
+    it('should throw SentryError when FirebaseRepository update fails', async () => {
+      mockFirebaseRepository.update.mockRejectedValueOnce(new Error('Firebase repository update failed'));
+
+      await expect(
+        firebaseGeneReviewService.updateReviewableContent(
+          'Genes/BRAF/mutations/0/description',
+          'old',
+          'new',
+          new Review(DEFAULT_USERNAME),
+          'uuid',
+        ),
+      ).rejects.toThrow(SentryError);
+    });
+  });
+
+  describe('acceptChanges', () => {
+    beforeEach(() => _beforeEach());
```

Comment:
Couldn't you put this logic at the `describe('Firebase Gene Review Service', () => {` level?

You can also be high brow about it. 🎩 

```ts
beforeEach(_beforeEach);
```